src/main/java/app/MainGameSearchApplication.java

```diff
@@ -11,25 +11,38 @@
+     *
+     * @param args command-line arguments
+     */
     public static void main(String[] args) {
-        // Initialize dependencies for FetchPlayerStatisticsInteractor
-        PlayerStatisticsRepositoryImpl playerStatisticsRepository = new PlayerStatisticsRepositoryImpl(); // Replace with actual implementation
-        FetchPlayerStatisticsInteractor fetchStatsBoundary = new FetchPlayerStatisticsInteractor(playerStatisticsRepository);
-        PlayGameController playGameController = PlayGameUseCaseFactory.createController();
+        // Step 1: Initialize dependencies for FetchPlayerStatisticsInteractor
+        final PlayerStatisticsRepositoryImpl playerStatisticsRepository = new PlayerStatisticsRepositoryImpl();
+        final FetchPlayerStatisticsInteractor fetchStatsBoundary =
+                new FetchPlayerStatisticsInteractor(playerStatisticsRepository);
+
+        // Step 2: Initialize PlayGameController
+        final PlayGameController playGameController = PlayGameUseCaseFactory.createController();
 
-        // Initialize dependencies for SearchInteractor
-        DBSearchDataAccessObject dbSearchDataAccess = new DBSearchDataAccessObject(); // Implements SearchDataAccessInterface
-        SearchViewModel searchViewModel = new SearchViewModel(); // Create an instance of SearchViewModel
-        SearchInteractor searchInteractor = new SearchInteractor(dbSearchDataAccess, searchViewModel); // Pass both dependencies
+        // Step 3: Initialize dependencies for SearchInteractor
+        final DBSearchDataAccessObject dbSearchDataAccess = new DBSearchDataAccessObject();
+        final SearchViewModel searchViewModel = new SearchViewModel();
+        final SearchInteractor searchInteractor = new SearchInteractor(dbSearchDataAccess, searchViewModel);
 
-        // Initialize dependencies for SearchHistoryController
-        SearchHistoryDataAccessObject searchHistoryDataAccess = new SearchHistoryDataAccessObject();
-        SearchHistoryInteractor searchHistoryInteractor = new SearchHistoryInteractor(searchHistoryDataAccess);
-        SearchHistoryController searchHistoryController = new SearchHistoryController(searchHistoryInteractor);
+        // Step 4: Initialize dependencies for SearchHistoryController
+        final SearchHistoryDataAccessObject searchHistoryDataAccess = new SearchHistoryDataAccessObject();
+        final SearchHistoryInteractor searchHistoryInteractor = new SearchHistoryInteractor(searchHistoryDataAccess);
+        final SearchHistoryController searchHistoryController = new SearchHistoryController(searchHistoryInteractor);
 
-        // Pass the correct arguments to MainFrame
-        MainFrame mainFrame = new MainFrame(playGameController, searchInteractor);
+        // Step 5: Pass the correct arguments to MainFrame and start the application
+        final MainFrame mainFrame = new MainFrame(playGameController, searchInteractor);
         mainFrame.start();
     }
```

Comment:
Ensure lifecycle management of all interactor dependencies (e.g., shutdown, cleanup).